packages/forms/src/index.ts

```diff
@@ -212,6 +212,87 @@ export const addPatternToPage = (
   };
 };
 
+export const movePatternBetweenPages = (
+  bp: Blueprint,
+  sourcePageId: PatternId,
+  targetPageId: PatternId,
+  patternId: PatternId,
+  position: string,
+  isPageMove: boolean = false
+): Blueprint => {
+  const sourcePage = bp.patterns[sourcePageId] as PagePattern;
+  const targetPage = bp.patterns[targetPageId] as PagePattern;
+
+  if (!sourcePage || !targetPage) {
+    throw new Error('Source or target page not found.');
+  }
+
+  if (sourcePage.type !== 'page' || targetPage.type !== 'page') {
+    throw new Error('Pattern is not a page.');
+  }
+
+  let updatedSourcePatterns: PatternId[];
+  let updatedTargetPatterns: PatternId[];
+
+  if (isPageMove) {
+    const sourcePagePatterns = sourcePage.data.patterns;
+    const indexToRemove = sourcePagePatterns.indexOf(patternId);
+
+    if (indexToRemove === -1) {
+      throw new Error(`Pattern ID ${patternId} not found in the source page.`);
+    }
+
+    const newPattern = sourcePagePatterns.splice(indexToRemove, 1).toString();
```

Comment:
`splice` in this case will modify the pattern array on the original pattern object, which we want to avoid. In the future, we might do things like put that object on an undo stack as a way to go back to a prior revision.

Can we just use slice to remove from the source page, and get a new array back? It seems like that's what's happening on the next section, so I'm wondering if this line is duplicating that?

Reply:
I ended up not using newPattern anymore and removed it.